services/QuillLMS/app/models/teacher_activity_feed.rb

```diff
@@ -13,18 +13,21 @@ def limit
   end
 
   def hydrate(ids:)
+    return [] if ids.empty?
+
     sessions = ActivitySession
-      .includes(:user, :activity, :classification, :classroom_unit)
```

Comment:
Why did you remove `:activity` here? Am I wrong in thinking that'll make the `session.activity.name` call below more expensive?

Reply:
So, I removed it because I believe it's redundant. The `:classification` relationship goes through `activity`, so having `:classification` here eager loads both `activity` and `activity_classification`. This finder performs these queries:

<img width="1669" alt="Screen Shot 2021-06-02 at 2 55 20 PM" src="https://user-images.githubusercontent.com/1304933/120538100-0fb54680-c3b4-11eb-90e9-218761ffa302.png">